app/domain/prerelease_for_platform.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# This consolidates the prelease checking logic into one place.
+# It only checks against specific platforms, not against the
+# general cases.
+class PrereleaseForPlatform
+  PYPI_PRERELEASE = /(a|b|rc|dev)[-_.]?[0-9]*$/.freeze
+
+  def self.prerelease?(version_number:, platform:)
+    new(version_number: version_number, platform: platform).prerelease?
+  end
+
+  def initialize(version_number:, platform:)
+    @version_number = version_number
+    @platform = platform.downcase
+  end
```

Comment:
Would it be simpler to have the class method only? I can't think of a reason we would need to hold on to an instance of this rather than just call the method directly which is what the code is doing now. Will there be other methods added in here?

Reply:
I think it can be class method-only.